quickstarts/microsoft.web/function-app-windows-consumption/main.bicep

```diff
@@ -0,0 +1,125 @@
+param _artifactsLocation string = deployment().properties.templateLink.uri
+
+@secure()
+@description('The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated. Use the defaultValue if the staging location is not secured.')
+param _artifactsLocationSasToken string = ''
+
+var packageUri = uri(_artifactsLocation, 'packages/nodejs.zip${_artifactsLocationSasToken}')
+
+var hostingPlanName = functionAppName
+var applicationInsightsName = functionAppName
+var storageAccountName = '${uniqueString(resourceGroup().id)}azfunctions'
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: storageAccountType
+  }
+  kind: 'Storage'
+}
+
+resource hostingPlan 'Microsoft.Web/serverfarms@2023-01-01' = {
+  name: hostingPlanName
+  location: location
+  sku: {
+    name: 'Y1'
+    tier: 'Dynamic'
+    size: 'Y1'
+    family: 'Y'
+  }
+}
+
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' = {
+  name: applicationInsightsName
+  location: location
+  tags: {
+    'hidden-link:${resourceId('Microsoft.Web/sites', applicationInsightsName)}': 'Resource'
+  }
+  properties: {
+    Application_Type: 'web'
+  }
+  kind: 'web'
+}
+
+resource functionApp 'Microsoft.Web/sites@2023-01-01' = {
+  name: functionAppName
+  location: location
+  kind: 'functionapp'
+  properties: {
+    serverFarmId: hostingPlan.id
+    siteConfig: {
+      appSettings: [
+        {
+          name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+          value: applicationInsights.properties.InstrumentationKey
+        }
+        {
+          name: 'AzureWebJobsStorage'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccountName};EndpointSuffix=${environment().suffixes.storage};AccountKey=${storageAccount.listKeys().keys[0].value}'
+        }
+        {
+          name: 'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccountName};EndpointSuffix=${environment().suffixes.storage};AccountKey=${storageAccount.listKeys().keys[0].value}'
+        }
+        {
+          name: 'WEBSITE_CONTENTSHARE'
+          value: toLower(functionAppName)
+        }
+        {
+          name: 'FUNCTIONS_EXTENSION_VERSION'
+          value: '~4'
+        }
+        {
+          name: 'FUNCTIONS_WORKER_RUNTIME'
+          value: functionWorkerRuntime
+        }
+        {
+          name: 'WEBSITE_NODE_DEFAULT_VERSION'
+          value: '~14'
+        }
+        {
+          name: 'WEBSITE_RUN_FROM_PACKAGE'
+          value: '1'
+        }
+      ]
+    }
+  }
+}
+
+resource msDeploy 'Microsoft.Web/sites/extensions@2023-01-01' = {
+  parent: functionApp
+  name: 'MSDeploy'
+  properties: {
+    packageUri: packageUri
+  }
+}
```

Comment:
According to @patelchandni, it sounds like the bicep-first approach might not work here since Functions requires a `zipdeploy` resource, which doesn't seem to yet be supported by Bicep. The PM for Functions deployments is @im-samz.

Reply:
Interesting, @ggailey777. Feel free to set time up next week to discuss this in more detail. Weird that Bicep wouldn't support the `zipdeploy` resource.